packages/react/kibana_context/root/root_provider.tsx

```diff
@@ -6,10 +6,16 @@
  * Side Public License, v 1.
  */
 
-import type { I18nStart } from '@kbn/core-i18n-browser';
-import type { AnalyticsServiceStart } from '@kbn/core-analytics-browser';
 import React, { FC, PropsWithChildren } from 'react';
 
+import type { AnalyticsServiceStart } from '@kbn/core-analytics-browser';
+import type { I18nStart } from '@kbn/core-i18n-browser';
+
+// @ts-expect-error EUI exports this component internally, but Kibana isn't picking it up its types
+import { useIsNestedEuiProvider } from '@elastic/eui/lib/components/provider/nested';
```

Comment:
@cee-chen Can this be something addressed in a future EUI release?  Or is this a mistake somehow, (I don't see how).

Reply:
Not a mistake, this has been done other places in Kibana as well. TBH I'm not 100% sure it's solvable on just the EUI level - I think it's a weird factor of how Kibana handles shared dependencies and its bazel pipeline. We might need to loop in KibanaOps for this as I'm not too sure why this happens either.